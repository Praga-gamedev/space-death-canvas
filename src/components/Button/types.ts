import { ButtonHTMLAttributes } from 'react';

export interface IButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {}
