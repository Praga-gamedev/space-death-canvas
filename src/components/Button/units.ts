import styled from '@emotion/styled';

import { colors } from 'src/colors';

import { IButtonProps } from './types';

export const S: Record<string, any> = {};

S.Button = styled.button`
    display: block;
    width: 250px;
    height: 50px;

    background-color: ${(props: IButtonProps) =>
        props.disabled ? colors.GrayScale_30 : 'transparent'};

    cursor: ${(props: IButtonProps) =>
        props.disabled ? 'not-allowed' : 'pointer'};

    border: 2px solid ${colors.secondary};
    border-radius: 15px;
    color: ${colors.GrayScale_0};
    transition: background-color 0.2s linear;

    &:hover:not(:disabled) {
        background-color: ${colors.secondary};
    }

    &:active:not(:disabled) {
        background-color: ${colors.secondaryAccent};
    }
`;
