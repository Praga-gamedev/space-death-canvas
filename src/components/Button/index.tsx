import React, { FC } from 'react';

import { StyledButton } from './units';
import { IButtonProps } from './types';

export const Button: FC<IButtonProps> = (props) => {
    return <StyledButton {...props} />;
};
