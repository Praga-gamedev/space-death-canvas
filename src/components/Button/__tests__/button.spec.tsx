import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Button } from '@components/Button';

describe('Button component', () => {
    test('button rendered correctly', () => {
        const children = 'Snapshot!';
        const ButtonComponent = renderer
            .create(<Button>{children}</Button>)
            .toJSON();
        expect(ButtonComponent).toMatchSnapshot();
    });

    test('text rendered correctly', () => {
        const buttonText = 'Click me';
        const ButtonComponent = mount(<Button>{buttonText}</Button>);

        expect(ButtonComponent.text()).toBe(buttonText);
    });

    test('content rendered correctly', () => {
        const buttonText = 'Click me';
        const ButtonComponent = mount(
            <Button>
                <span>{buttonText}</span>
            </Button>
        );

        expect(ButtonComponent.find('span')).toHaveLength(1);
        expect(ButtonComponent.find('span').at(0).text()).toBe(buttonText);
    });

    test('click event working', () => {
        const onClick = jest.fn();
        const props = { onClick };

        const ButtonComponent = mount(<Button {...props} />);
        ButtonComponent.simulate('click');

        expect(onClick).toBeCalledTimes(1);
    });
});
