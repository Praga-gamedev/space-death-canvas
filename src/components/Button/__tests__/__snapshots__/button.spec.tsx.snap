// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component button rendered correctly 1`] = `
<button
  className="css-1a7nukv"
>
  Snapshot!
</button>
`;
